File: packages/permissions/ios/Plugin/Classes/Results/PermissionStatus.swift
import Foundation
import Capacitor

@objc public class PermissionStatus: NSObject, Result {
    let permission: Permission
    let state: PermissionState

    init(permission: Permission, state: PermissionState) {
        self.permission = permission
        self.state = state
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["permission"] = permission.rawValue
        result["state"] = state.rawValue
        return result as AnyObject
    }
}

File: packages/permissions/android/src/main/java/io/capawesome/capacitorjs/plugins/permissions/classes/results/PermissionStatus.java
package io.capawesome.capacitorjs.plugins.permissions.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.permissions.enums.PermissionState;
import io.capawesome.capacitorjs.plugins.permissions.enums.PermissionType;
import io.capawesome.capacitorjs.plugins.permissions.interfaces.Result;

public class PermissionStatus implements Result {

    @NonNull
    private final PermissionType permission;

    @NonNull
    private final PermissionState state;

    public PermissionStatus(@NonNull PermissionType permission, @NonNull PermissionState state) {
        this.permission = permission;
        this.state = state;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("permission", permission.getValue());
        result.put("state", state.getValue());
        return result;
    }
}
